livy_notebooks/Tornado/Tornado_Clustering_PySpark_Viz.ipynb

```diff
@@ -0,0 +1,1902 @@
+      ],
+      "text/plain": [
+       "<IPython.core.display.HTML object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "SparkSession available as 'spark'.\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "from pyspark.ml.feature import VectorAssembler\n",
+    "from pyspark.ml.clustering import KMeans"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Preparation of the input data\n",
+    "\n",
+    "The following cell uses the ibmdbpy library to establish a connection from this notebook to the database. The cell prepares the input data based on sample tables that are available in every Db2 Warehouse container out of the box.\n",
+    "\n",
+    "Here we check if the tables containing data about tornados and home insurance holders have already been created, if not we build them."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "FloatProgress(value=0.0, bar_style='info', description='Progress:', layout=Layout(height='25px', width='50%'),…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Connection closed."
+     ]
+    }
+   ],
+   "source": [
+    "from ibmdbpy import IdaDataBase\n",
+    "idadb =IdaDataBase('bludb')\n",
+    "\n",
+    "# create table if does not exist already\n",
+    "\n",
+    "if idadb.exists_table(\"TORNADO\") == False:\n",
+    "    statement1 = idadb.ida_query('create table tornado as (select objectid, yr, mo, dy, time, '+\n",
```

Comment:
This might be OK for your use case, but please not that running CTAS statement via `ida_query` can lead to problems (#43).